src/esp/physics/PhysicsManager.cpp

```diff
@@ -3,16 +3,28 @@
 // LICENSE file in the root directory of this source tree.
 
 #include "PhysicsManager.h"
-#include "esp/assets/CollisionMeshData.h"
-
 #include <Magnum/Math/Range.h>
-
+#include "esp/assets/CollisionMeshData.h"
+#include "esp/physics/objectManagers/RigidObjectManager.h"
+#include "esp/sim/Simulator.h"
 namespace esp {
 namespace physics {
 
-bool PhysicsManager::initPhysics(scene::SceneNode* node) {
+PhysicsManager::PhysicsManager(
+    assets::ResourceManager& _resourceManager,
+    const metadata::attributes::PhysicsManagerAttributes::cptr&
+        _physicsManagerAttributes)
+    : resourceManager_(_resourceManager),
+      physicsManagerAttributes_(_physicsManagerAttributes),
+      rigidObjectManager_(RigidObjectManager::create()) {}
+
+bool PhysicsManager::initPhysics(
+    scene::SceneNode* node,
+    const std::shared_ptr<esp::physics::PhysicsManager>& physMgr) {
   physicsNode_ = node;
-
+  // set the rigidObjectManager's weak reference to physics manager tp be based
+  // on the same shared pointer that Simulator is using.
+  rigidObjectManager_->setPhysicsManager(physMgr);
```

Comment:
I was passing Simulator's pointer around because I wanted to always have that specific pointer (to stay synched with Simulator).  I don't think having 2 shared pointers pointing at same PM would behave the same way, would it? If sim->physMgrPtr starts pointing at another physMgr, rigidObjectMgr->weakPhysMgr will not be pointing at the same thing, will it?

Reply:
`shared_from_this()` will grab that same shared pointer

Reply:
Cool, I didn't know that.